### DISCUSSION TITLE
iNav 4.1 GPS QMC5883 issue

### DISCUSSION THREAD

**Discussion by MaximMalkov** [iNavFlight/inav#8019] [Q&A] [answered]
I have two quads: small 4 inch and a bit bigger 7 inch. On both of them the same GPS is installed - Mateksys QMC5883 in almost the same manner. Both GPS are working and finds satelites well. The first quad 4 inch is navigating well and when I swith it to Alt Hold - it is holding the altitude well. But on the second one 7 inch quad - when I swith to Alt Hold - it is starts rotating and fall down immediately. GPS is positioned almost in the same manner on both quads - with pins to the front, with the following alignment: 

4 inch quad:
set align_mag = CW0
set mag_declination = 0
set align_mag_roll = 300
set align_mag_pitch = 1800
set align_mag_yaw = 900

results of calibration: 
set magzero_x = -49
set magzero_y = -103
set magzero_z = -311
set maggain_x = 1658
set maggain_y = 1669
set maggain_z = 1696

7 inch quad:
set align_mag = CW0
set mag_declination = 0
set align_mag_roll = 210 (it has less angle)
set align_mag_pitch = 1800
set align_mag_yaw = 900

results of calibration:
set magzero_x = -228
set magzero_y = 61
set magzero_z = -1255
set maggain_x = 1779
set maggain_y = 1721
set maggain_z = 1623

I wonder, how it could be, that on the first quad it works, and on the second one does not? Any one could have any ideas?

**Comment by Jetrell** [comment, marked as answer]
Main things to look for are - Incorrect magnetometer/flight control board arrow alignment, with respect to each others. 
Or the magnetometer calibration is off, due to interference from something on the quad.

**Reply by MaximMalkov** [reply]
Thank you for the ideas, I will check it
